### PR TITLE
Feat/implement as

### PR DESCRIPTION
**Pull request by sum2000**
adds support for #76

### CODE REVIEW

File: eris_test.go [hunk 1]
```diff
@@ -363,6 +363,128 @@ func TestErrorIs(t *testing.T) {
...
+		"wrapped error against different wrapped error": {
+			cause:  eris.Wrap(nil, "some other error"),
+			target: &wrappedErr,
+			match:  false,
+			output: nil,
+		},
+		"custom error against similar type error": {
+			cause:  customErr,
+			target: &withLayer{msg: "additional layer", err: externalError},
+			match:  true,
```
Comment (morningvera) [inline eris_test.go:466]: why does this case return true? seems like it should be false but i'm probably missing something

Comment (sum2000) [reply eris_test.go:466]: I added custom As implementation for customError that's why, see the top of test file for that.

File: eris.go [hunk 2]
```diff
@@ -176,6 +212,17 @@ func (e *rootError) Is(target error) bool {
 	return e.msg == target.Error()
 }
 
+func (e *rootError) As(target interface{}) bool {
+	t := reflect.Indirect(reflect.ValueOf(target)).Interface()
```
[the remark this answers is not in the document]
Comment (sum2000) [reply eris.go:216]: I will add a comment, basically dereferencing the pointer target and getting the value the pointer is holding as an interface

File: eris.go [hunk 3]
```diff
@@ -126,6 +126,42 @@ func Is(err, target error) bool {
...
+	// target must be a non-nil pointer
+	if typ.Kind() != reflect.Ptr || val.IsNil() {
+		return false
+	}
+	// *target must be interface or implement error
+	if e := typ.Elem(); e.Kind() != reflect.Interface && !e.Implements(reflect.TypeOf((*error)(nil)).Elem()) {
+		return false
+	}
+	errType := reflect.TypeOf(err)
+	if errType != reflect.TypeOf(&wrapError{}) && errType != reflect.TypeOf(&rootError{}) && reflect.TypeOf(err).AssignableTo(typ.Elem()) {
```
[the remark this answers is not in the document]
Comment (sum2000) [reply eris.go:151]: missed it, good catch

File: eris.go [hunk 4]
```diff
@@ -126,6 +126,42 @@ func Is(err, target error) bool {
...
+// As finds the first error in err's chain that matches target, and if so, sets target to that error value and returns true.
+// Otherwise, it returns false.
+//
+// The chain consists of err itself followed by the sequence of errors obtained by repeatedly calling Unwrap.
+//
+// An error matches target if the error's concrete value is assignable to the value pointed to by target,
+// or if the error has a method As(interface{}) bool such that As(target) returns true.
+func As(err error, target interface{}) bool {
+	if target == nil || err == nil {
+		return err == target
```
[the remark this answers is not in the document]
Comment (sum2000) [reply eris.go:138]: I am okay with returning false, not sure assigning makes sense as one of them will be nil. golang As just panics in this case

[the remark this answers is not in the document]
Comment (morningvera) [reply eris.go:138]: yeah, agreed that we don't need to assign it for this case

File: eris_test.go (hunk 1 above)
Comment (morningvera) [reply eris_test.go:466]: hmm, i'm not seeing that in the diff though

**Comment by morningvera** [comment]
@sum2000 any progress on this?

**Comment by sum2000** [comment]
Not yet, are we planning on another release? I'll like to postpone it until then. Preparing for certification in extra time

**Comment by morningvera** [comment]
closes #76

**Pull request merged**